agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2080,13 +2080,17 @@ bool backhaul_manager::handle_1905_1_message(ieee1905_1::CmduMessageRx &cmdu_rx,
     }
 }
 
-bool backhaul_manager::get_sta_bssid(const sMacAddr &sta_mac, sMacAddr &bssid)
+bool backhaul_manager::get_sta_bssid_and_socket(const sMacAddr &sta_mac, sMacAddr &bssid,
+                                                Socket **socket)
 {
     for (const auto &slave : slaves_sockets) {
         auto associated_clients_map = slave->associated_clients_map;
         for (const auto &vap : associated_clients_map) {
             if (vap.second.find(sta_mac) != vap.second.end()) {
                 bssid = vap.first;
+                if (socket) {
+                    *socket = slave->slave;
+                }
```

Comment:
Assuming a reference to a pointer:
```suggestion
                socket = slave->slave;
```